lib/babe/babe.go

```diff
@@ -446,6 +436,32 @@ func (b *Service) invokeBlockAuthoring(epoch uint64) error {
 	}
 }
 
+func (b *Service) waitForEpochStart(epoch uint64) (uint64, error) {
+	// get start slot for current epoch
+	epochStart, err := b.epochState.GetStartSlotForEpoch(epoch)
+	if err != nil {
+		logger.Error("failed to get start slot for current epoch", "epoch", epoch, "error", err)
+		return 0, err
+	}
+
+	epochStartTime := getSlotStartTime(epochStart, b.slotDuration)
```

Comment:
nit: you could make `getSlotStartTime` a receiver function on `Service` and can omit the `slotDuration` param.

Reply:
I kinda like it like this cause it seems more generic, but happy to change it if you think it's better as a receiver